### PR TITLE
Task 12

### PR DESCRIPTION
**Pull request by alenakuznetsova**
merge old and new version for viewing demo

### CODE REVIEW

**Comment by alenakuznetsova** [comment]
create routing (login, registration)

# Task 12
[Demo](https://kottans-frontend-2018.github.io/assignments_alenakuznetsova/task_12/dist/#login)
[Source](https://github.com/kottans-frontend-2018/assignments_alenakuznetsova/tree/task_12)

**Comment by zonzujiro** [comment]
It is a very big PR. I mean... really. Can you make new PR with only new files, without bundles, and without removed files?

**Comment by alenakuznetsova** [comment]
Thank you, I'll try

**Comment by suchov** [comment]
It looks like a new record! 💥

**Comment by alenakuznetsova** [comment]
sorry, I only have been studying )

**Comment by viktor-yakubiv** [comment]
Don't sorry. I've already said: you have been studying 🙂

We a learning much too, as well.

**Comment by suchov** [comment]
no worries @alenakuznetsova all is great, and it's great that you can do mistakes with us - you'll remember them and it'll be easier in the future projects ;)

File: task_12/dist/app.bundle.js [hunk 1]
```diff
@@ -1630,6 +1607,112 @@ class Time {
...
+
+    isAuthorized(){
+        if( !this.tokenIsNotExpired() ){
+            this.clearStorage();
+            return false;
+        }
+        return true;
+    }
+
+    tokenIsNotExpired(){
```
Comment (AMashoshyna) [inline task_12/dist/app.bundle.js:1654]: for this function to return `true`, you need two conditions to be met: `this.token` must exist and `this.claims.exp * 1000` must be greater than Date.now(). All other combinations of values should return `false`. So what you can do is like so:
` tokenIsNotExpired() {
return this.token && this.claims.exp * 1000 > Date.now();
}`

File: task_12/dist/app.bundle.js [hunk 2]
```diff
@@ -1630,6 +1607,112 @@ class Time {
...
+        this._token = null;
+        this._claims = null;
+        localStorage.removeItem('token');
+        localStorage.removeItem('claims');
+    }
+
+    isAuthorized(){
+        if( !this.tokenIsNotExpired() ){
+            this.clearStorage();
+            return false;
```
Comment (AMashoshyna) [inline task_12/dist/app.bundle.js:1649]: it is not a good pattern to write boolean value to be returned explicitly. Try to rewrite this function so, that it returns the result of some evaluation directly. Start function code with `return`

Example
Not good:
```
if (somecondition.isTrue()) {
return true;
} else {
return false;
}
```
Better:

```return somecondition.isTrue()```

**Pull request closed without merging**